langchain/src/document_loaders/web/firecrawl.ts

```diff
@@ -0,0 +1,96 @@
+import FirecrawlApp from "@mendable/firecrawl-js";
+import { Document } from "@langchain/core/documents";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import { BaseDocumentLoader } from "../base.js";
+
+/**
+ * Interface representing the parameters for the Firecrawl loader. It
+ * includes properties such as the URL to scrape or crawl and the API key.
+ */
+interface FirecrawlLoaderParameters {
+  /**
+   * URL to scrape or crawl
+   */
+  url: string;
+  apiKey?: string;
+  mode?: "crawl" | "scrape";
+  params?: Record<string, unknown>;
+}
+interface FirecrawlDocument {
+  markdown: string;
+  metadata: Record<string, unknown>;
+}
+
+/**
+ * Class representing a document loader for loading data from
+ * Firecrawl (firecrawl.dev). It extends the BaseDocumentLoader class.
+ * @example
+ * ```typescript
+ * const loader = new FireCrawlLoader({ url: "{url}", apiKey: "{apiKey}", mode: "crawl" });
+ * const docs = await loader.load();
```

Comment:
```suggestion
 * ```typescript
 * const loader = new FireCrawlLoader({
 *   url: "{url}",
 *   apiKey: "{apiKey}",
 *   mode: "crawl"
 * });
 * const docs = await loader.load();
```
will look better in the api refs if it has formatting